src/scenes/Partners/Detail/Tabs/Projects/PartnerDetailProjects.tsx

```diff
@@ -1,13 +1,66 @@
+import { useQuery } from '@apollo/client';
 import { Box, Typography } from '@mui/material';
-import { PartnerDetailsFragment } from '../../PartnerDetail.graphql';
+import { GridSortModel } from '@mui/x-data-grid';
+import { useCallback, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Order } from '~/api/schema/schema.graphql';
 import { PartnerDetailProjectsTable } from './PartnerDetailProjectsTable';
+import { PartnerProjectsDocument } from './PartnerProjects.graphql';
 
-interface Props {
-  partner: PartnerDetailsFragment | undefined;
-}
+export const PartnerDetailProjects = () => {
+  const { partnerId = '' } = useParams();
+  const [page, setPage] = useState<number>(0);
+  const [sortModel, setSortModel] = useState<GridSortModel>([
+    { field: 'name', sort: 'asc' },
+  ]);
 
-export const PartnerDetailProjects = ({ partner }: Props) => {
-  const projects = partner?.projects;
+  const { data, refetch } = useQuery(PartnerProjectsDocument, {
+    variables: {
+      id: partnerId,
+      projectListInput: {
+        count: 25,
+      },
+    },
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: 'no-cache',
```

Comment:
@CarsonF 

When adding sorting I got the following warning in the console:

Cache data may be lost when replacing the partner field of a Query object. This could cause additional (usually avoidable) network requests to fetch data that were otherwise cached. To address this problem (which is not a bug in Apollo Client), either ensure all objects of type Partner have an ID or a custom merge function, or define a custom merge function for the Query.partner field, so InMemoryCache can safely merge these objects

I am pretty sure this is because Apollo is trying to merge the new list of items, but because it is a sort some of the items are the same just in different positions so they are being seen as potential duplicates... I went in to typePolicies.base.ts and tried to add some logic that should handle sort like so:

```
  Partner: {
    fields: {
      projects: {
        keyArgs: [['projectListInput', ['page', 'count', 'sort', 'order']]],
        merge(
          existing: SecuredProjectList | undefined,
          incoming: SecuredProjectList,
          { readField }
        ) {
          // Pagination logic: If the id of the last item of existing is not equal to the id of the first item of incoming,
          // that means incoming data is a new page of data, so concatenate it.
          const existingLastItem = existing && last(existing.items);
          const incomingFirstItem = first(incoming.items);
          const existingLastItemId = existingLastItem && readField('id', existingLastItem);
          const incomingFirstItemId = incomingFirstItem && readField('id', incomingFirstItem);
          if (existingLastItemId !== incomingFirstItemId) {
            // Here, we spread the existing and incoming fields to maintain other properties of SecuredProjectList
            return {
              ...existing,
              ...incoming,
              items: [...(existing?.items || []), ...incoming.items],
            };
          }
          // Otherwise, the incoming data is an update operation, so replace the existing one.
          return incoming;
        }
      },
    },
  },
```

But that would not fix the issue. I do not think busting the cache here is ideal but I don't see any noticable performance impact (probably beacause of the pagination). Since you are more experienced with Apollo (and I noticed that you have added before to typePolicies) I'd love a better solution if you have one!